posthog/hogql_queries/insights/test/test_trends_query_runner.py

```diff
@@ -0,0 +1,280 @@
+from typing import List, Optional
+from freezegun import freeze_time
+from pydantic import BaseModel
+
+from posthog.hogql_queries.insights.trends_query_runner import TrendsQueryRunner
+from posthog.schema import DateRange, EventsNode, IntervalType, TrendsFilter, TrendsQuery
+from posthog.test.base import APIBaseTest, ClickhouseTestMixin, _create_event, _create_person
+
+
+class Series(BaseModel):
+    event: str
+    timestamps: List[str]
+
+
+class SeriesTestData(BaseModel):
```

Comment:
pydantic seems a bit overkill for tests, I think we use dataclasses typically

Reply:
Ooh, didn't know about this, thanks! Do you have an example of how to replicate this using dataclasses (or a pointer to some code using dataclasses?)

Reply:
Pydantic runs a lot of inline validations, making object creation slow. We used it for all AST nodes, and swapping those with dataclasses saw the HogQL part for heavy queries drop from ~40s to ~10s (and Michael's latest work will take that to <1s). So it's worth using pydantic when that matters, but defaulting to dataclasses.

Search for `dataclass` in the codebase for some examples, but tldr:

```py
from dataclasses import dataclass

@dataclass
class Person():
    name: str
    age: int
    height: float
    email: str
```

Now you can do `Person(name='asd', age=12, etc)`. Without it, you'd have to write a constructor and do something like:

```py
from dataclasses import dataclass

class Person():
    name: str
    age: int
    height: float
    email: str

    def __init__(name, age, height, email):
        self.name = name
        self.age = age
        self.height = height
        self.email = email
```

Reply:
Amazing, thank you!